File: Python/Wieża i piony.py
M, N = map(int, input().split())
board = [input() for _ in range(M)]
row_counts = [0] * M
col_counts = [0] * N
for i in range(M):
    for j in range(N):
        if board[i][j] == '#':
            row_counts[i] += 1
            col_counts[j] += 1
max_attacked = 0
for i in range(M):
    for j in range(N):
        if board[i][j] == '.':
            max_attacked = max(max_attacked, row_counts[i] + col_counts[j])
print(max_attacked)

File: Java/Wieża i piony.java
import java.util.Scanner;

public class Main 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int M = input.nextInt();
        int N = input.nextInt();
        input.nextLine();
        String[] board = new String[M];
        for (int i = 0; i < M; i++)
            board[i] = input.nextLine();
        int[] rowCounts = new int[M];
        int[] colCounts = new int[N];
        for (int i = 0; i < M; i++)
            for (int j = 0; j < N; j++)
                if (board[i].charAt(j) == '#') 
                {
                    rowCounts[i]++;
                    colCounts[j]++;
                }
        int maxAttacked = 0;
        for (int i = 0; i < M; i++)
            for (int j = 0; j < N; j++)
                if (board[i].charAt(j) == '.') 
                    maxAttacked = Math.max(maxAttacked, rowCounts[i] + colCounts[j]);
        System.out.print(maxAttacked);
        input.close();
    }
}